Bug list:

Error to fix:


[2015-03-04 09:07:53,821] [display 1] - Bot Id: 1 (143:15:02)
`BTC-e' Total execution time: 10.1ms (min:6.6ms; max:196.6ms)
`BTC-e' Pair update execution time: 2.7ms (min:1.6ms; max:186.1ms)
`BTC-e' Algorithm `triangularArbitrage' execution time: 4.9ms (min:3.7ms; max:130.9ms)
`BTC-e' at t=1425460072 - Total estimated value 99.077858 USD (+3.13%)
PPC:    0.000000
USD:    10.357193
BTC:    0.000753
LTC:    9.489282
NVC:    14.071800
NMC:    9.299114
GBP:    7.720106        (+6.722980)
RUR:    677.260169      (+625.974350)
CNH:    60.836866
EUR:    2.726995
Active Orders:
`BTC-e' id:367  pending         <buy LTC/CNH rate:0.09 amount:60.82>
`BTC-e' id:387  pending         <buy LTC/CNH rate:0.09 amount:60.82>
`BTC-e' id:389  pending         <buy LTC/CNH rate:0.09 amount:60.82>
`BTC-e' id:409  pending         <buy BTC/GBP rate:0.00657 amount:6.72298>
`BTC-e' id:459  pending         <sell USD/RUR rate:60.451 amount:10.42451>
`BTC-e' id:470  pending         <buy LTC/RUR rate:0.00897 amount:625.97435>

[2015-03-04 09:07:54,999] [error 0] - This order is incomplete `<sell GBP/USD rate:None amount:None>'.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 269, in taskBalance
    string = string + self.printBalance()
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 209, in printBalance
    value = self.estimateValue(ex)
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 198, in estimateValue
    value = value + order.updateChain().estimateChain(balance[currency])
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 413, in estimateChain
    amount = order.estimate(amount, mode)
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 392, in estimate
    info = self.getInfo(amount)
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 336, in getInfo
    raise error("This order is incomplete `%s'." % (str(self)))
error: This order is incomplete `<sell GBP/USD rate:None amount:None>'.



---------------


[2015-03-11 10:44:45,060] [display 1] - Bot Id: 1 (2:18:29)
`BTC-e' Total execution time: 10.5ms (min:6.7ms; max:33.2ms)
`BTC-e' Pair update execution time: 5.2ms (min:2.6ms; max:27.1ms)
`BTC-e' Algorithm `triangularArbitrage' execution time: 5.1ms (min:3.9ms; max:14.9ms)
`BTC-e' at t=1426070684 - Total estimated value 100.956859 USD (+0.00%)
PPC:    0.000000
USD:    100.956849
BTC:    0.000003
LTC:    0.000000
NVC:    0.000000
NMC:    0.000000
GBP:    0.000010
RUR:    0.000771
CNH:    0.000000
EUR:    0.000000
Active Orders: None
[2015-03-11 10:44:46,190] [display 1] - Bot Id: 1 (2:18:30)
`BTC-e' Total execution time: 10.2ms (min:6.7ms; max:33.2ms)
`BTC-e' Pair update execution time: 4.3ms (min:2.6ms; max:27.1ms)
`BTC-e' Algorithm `triangularArbitrage' execution time: 5.7ms (min:3.9ms; max:14.9ms)
`BTC-e' at t=1426070684 - Total estimated value 100.956859 USD (+0.00%)
PPC:    0.000000
USD:    100.956849
BTC:    0.000003
LTC:    0.000000
NVC:    0.000000
NMC:    0.000000
GBP:    0.000010
RUR:    0.000771
CNH:    0.000000
EUR:    0.000000
Active Orders: None

Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 269, in taskBalance
    string = string + self.printBalance()
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 209, in printBalance
    value = self.estimateValue(ex)
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 198, in estimateValue
    value = value + order.updateChain().estimateChain(balance[currency])
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 181, in updateChain
    order.update(mode)
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 171, in update
    self.rate = self.getUpdatedRate(pair, mode)
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 571, in getUpdatedRate
    return 1. / pair.getAsk()
TypeError: unsupported operand type(s) for /: 'float' and 'NoneType'

[2015-03-11 10:58:18,784] [error 0] - The read operation timed out
[2015-03-11 11:01:18,044] [error 0] - The read operation timed out
[2015-03-11 11:03:35,128] [error 0] - The read operation timed out
[2015-03-11 11:12:33,239] [error 0] - The read operation timed out
[2015-03-11 11:17:28,269] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 502: Bad Gateway
[2015-03-11 11:17:28,270] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 502: Bad Gateway
[2015-03-11 11:30:59,264] [error 0] - The read operation timed out
[2015-03-11 11:32:31,743] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 11:32:31,743] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 11:36:54,057] [error 0] - The read operation timed out
[2015-03-11 11:44:17,452] [error 0] - The read operation timed out
[2015-03-11 11:45:29,205] [error 0] - The read operation timed out
[2015-03-11 11:47:13,826] [error 0] - The read operation timed out
[2015-03-11 11:52:14,309] [error 0] - The read operation timed out
[2015-03-11 11:53:09,180] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 11:53:09,180] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 11:54:43,215] [error 0] - The read operation timed out
[2015-03-11 11:56:19,821] [error 0] - The read operation timed out
[2015-03-11 12:14:22,223] [error 0] - The read operation timed out
[2015-03-11 12:18:42,162] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 502: Bad Gateway
[2015-03-11 12:18:42,163] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 502: Bad Gateway
[2015-03-11 12:21:10,492] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:21:10,492] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:21:27,956] [error 0] - The read operation timed out
[2015-03-11 12:38:47,876] [error 0] - The read operation timed out
[2015-03-11 12:41:54,423] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:41:54,424] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:44:16,298] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:44:16,299] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:46:32,684] [error 0] - The read operation timed out
[2015-03-11 12:47:16,144] [error 0] - The read operation timed out
[2015-03-11 12:47:25,821] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:47:25,821] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:48:24,408] [error 0] - The read operation timed out
[2015-03-11 12:52:31,056] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 12:52:31,057] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 13:02:29,660] [error 0] - The read operation timed out
[2015-03-11 13:02:50,083] [error 0] - The read operation timed out
[2015-03-11 13:05:17,077] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 13:05:17,078] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 13:05:39,180] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 13:05:39,181] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-11 13:07:09,130] [error 0] - The read operation timed out


-----------------------


[2015-03-13 12:17:24,732] [display 1] - Bot Id: 1 (28:44:57)
`BTC-e' Total execution time: 12.2ms (min:6.6ms; max:242.9ms)
`BTC-e' Pair update execution time: 4.5ms (min:2.4ms; max:234.5ms)
`BTC-e' Algorithm `triangularArbitrage' execution time: 4.4ms (min:3.8ms; max:36.0ms)
`BTC-e' at t=1426249042 - Total estimated value 101.057341 USD (+0.10%)
PPC:    0.000000
USD:    71.034087
BTC:    0.000003
LTC:    4.979797        (+4.107430)
NVC:    0.000000
NMC:    0.000000
GBP:    0.000010
RUR:    724.428751
CNH:    0.000000
EUR:    0.000000
Active Orders:
`BTC-e' id:447  pending         <sell LTC/RUR rate:123.6001 amount:4.10743>

[2015-03-13 12:17:25,976] [error 0] - This order is incomplete `<sell EUR/USD rate:None amount:None>'.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 269, in taskBalance
    string = string + self.printBalance()
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 209, in printBalance
    value = self.estimateValue(ex)
  File "/home/blaise/bitcoin_arbitrage/bot.py", line 198, in estimateValue
    value = value + order.updateChain().estimateChain(balance[currency])
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 413, in estimateChain
    amount = order.estimate(amount, mode)
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 392, in estimate
    info = self.getInfo(amount)
  File "/home/blaise/bitcoin_arbitrage/exchanges/order.py", line 336, in getInfo
    raise error("This order is incomplete `%s'." % (str(self)))
error: This order is incomplete `<sell EUR/USD rate:None amount:None>'.

[2015-03-13 12:17:28,882] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad Request
[2015-03-13 12:17:28,883] [error 0] - Error while retreiving the page `https://btc-e.com/api/3/ticker/ltc_gbp-ltc_rur-btc_gbp-ltc_cnh-nmc_usd-ppc_usd-nmc_btc-eur_rur-eur_usd-ltc_usd-nvc_btc-usd_rur-ltc_eur-btc_usd-ltc_btc-ppc_btc-btc_cnh-btc_eur-nvc_usd-gbp_usd-usd_cnh-btc_rur' (timeout [10s]): HTTPError(): HTTP Error 400: Bad R
equest

